pkg/config/structure/unmarshal.go

```diff
@@ -249,6 +250,9 @@ func (n *leafNodeImpl) ChildrenKeys() ([]string, error) {
 func (n *leafNodeImpl) GetBool() (bool, error) {
 	if n.val.Kind() == reflect.Bool {
 		return n.val.Bool(), nil
+	} else if n.val.Kind() == reflect.Int {
+		stringVal := strconv.FormatInt(n.val.Int(), 10)
```

Comment:
I don't think it's necessary to convert to a string just to reuse `convertToBool`. None of the values that `convertToBool` care about will result from this conversion aside from the int case. Instead, I'd change this to:

`return n.val.Int() != 0`

Reply:
`!= 0` is more c-esque but will allow something like `42` to be truthy, is that alright?

Reply:
fixed in https://github.com/DataDog/datadog-agent/pull/29604/commits/924d6e2cd632bf0c47aac596c0b7438663ee8280